src/code/z_vimode.c

```diff
@@ -1,23 +1,378 @@
+            {
+                0x0404233A,
+                0x00000270,
+                0x00150C69,
+                0x0C6F0C6E,
+                0x00800300,
+                0x005F0239,
+                0x0009026B,
+            },
+            {
+                0x03E52239,
+                0x0000020C,
+                0x00000C15,
+                0x0C150C15,
+                0x006C02EC,
+                0x002501FF,
+                0x000E0204,
+            },
+            {
+                0x04651E39,
+                0x0000020C,
+                0x00000C10,
+                0x0C1C0C1C,
+                0x006C02EC,
+                0x002501FF,
+                0x000E0204,
+            },
+        };
+        const ViModeStruct* ptr = &D_801DF0C0[type];
+
+        viMode->customViMode.comRegs.burst = ptr->unk_00;
+        viMode->customViMode.comRegs.vSync = ptr->unk_04;
+        viMode->customViMode.comRegs.hSync = ptr->unk_08;
+        viMode->customViMode.comRegs.leap = ptr->unk_0C;
+        viMode->customViMode.comRegs.hStart = ptr->unk_10;
+        viMode->customViMode.fldRegs[0].vStart = ptr->unk_14;
+        viMode->customViMode.fldRegs[1].vStart = ptr->unk_14;
+        viMode->customViMode.fldRegs[0].vBurst = ptr->unk_18;
+        viMode->customViMode.fldRegs[1].vBurst = ptr->unk_18;
+    } else {
+        __assert("../z_vimode.c", 216);
+    }
+
+    viMode->customViMode.comRegs.hStart += (unk_left << 16) + (s16)unk_right;
+    viMode->customViMode.fldRegs[0].vStart += (unk_top << 16) + (s16)unk_bottom;
+    viMode->customViMode.fldRegs[1].vStart += (unk_top << 16) + (s16)unk_bottom;
+
+    if (cond_44) {
+        viMode->customViMode.comRegs.vSync++;
+        if (type == 2) {
+            viMode->customViMode.comRegs.hSync += 0x40001;
+            viMode->customViMode.comRegs.leap += 0xFFFCFFFE;
+        }
+    } else {
+        viMode->customViMode.fldRegs[0].vStart += 0xFFFDFFFE;
+        if (type == 2) {
+            viMode->customViMode.fldRegs[0].vBurst += 0xFFFCFFFE;
+        } else if (type == 0) {
+            viMode->customViMode.fldRegs[1].vBurst += 0x2FFFE;
+        }
+    }
+
+    viMode->customViMode.comRegs.xScale = (SCREEN_WIDTH << 10) / SCREEN_WIDTH;
+    viMode->customViMode.comRegs.vCurrent = 0;
+
+    viMode->customViMode.fldRegs[0].origin = width * 2 * (unk_7C ? 1 : 2);
+    viMode->customViMode.fldRegs[1].origin = width * 2 * (unk_7C ? 1 : 2) * (unk_70 ? 1 : 2);
+
+    viMode->customViMode.fldRegs[0].yScale = yScaleLo | yScaleHi0;
+    viMode->customViMode.fldRegs[1].yScale = yScaleLo | yScaleHi1;
+
+    viMode->customViMode.fldRegs[0].vIntr = 2;
+    viMode->customViMode.fldRegs[1].vIntr = 2;
+}
+
+void ViMode_Save(ViMode* viMode) {
+    SREG(48) = viMode->viModeBase;
+    SREG(49) = viMode->viWidth;
+    SREG(50) = viMode->viHeight;
+    SREG(51) = viMode->unk_64;
+    SREG(52) = viMode->unk_60;
+    SREG(53) = viMode->unk_5C;
+    SREG(54) = viMode->unk_58;
+
+    if (SREG(58) == 1) {
+        SREG(58) = 0;
```

Comment:
Do you know if there's any context for `SREG(58)`. I wonder if this is boolean or not